File: lecture04/01_palindromic/template.cpp
// palindromic.cpp

#include <iostream>
#include <string>
#include <unordered_set>

using namespace std;

// Functions

bool is_palindromic(string &s) {
    // TODO: Determine if string s contains a palindromic permutation
    return false;
}

// Main execution

int main(int argc, char *argv[]) {
    string word;

    while (cin >> word) {
    	cout << (is_palindromic(word) ? "Yes" : "No") << endl;
    }

    return 0;
}

File: lecture04/01_palindromic/template.py
#!/usr/bin/env python3

import collections
import sys

# Functions

def is_palindromic(s):
    # TODO: Determine if string s contains a palindromic permutation
    return False

# Main Execution

if __name__ == '__main__':
    for word in sys.stdin:
        print('Yes' if is_palindromic(word.rstrip()) else 'No')
